File: read_excel.py
import pandas as pd
import json

file_path = r'C:\Users\jyaxo\Desktop\nomina-v2\nomina-\BONOS  ABRIL 2026.xlsx'
xls = pd.ExcelFile(file_path)

data = {}
for sheet in xls.sheet_names:
    df = pd.read_excel(file_path, sheet_name=sheet)
    data[sheet] = df.head(10).to_dict(orient='records')

with open('excel_data.json', 'w', encoding='utf-8') as f:
    json.dump(data, f, ensure_ascii=False, indent=2)

File: backend/read_excel.js
const XLSX = require('xlsx');
const fs = require('fs');

const workbook = XLSX.readFile('C:/Users/jyaxo/Desktop/nomina-v2/nomina-/BONOS  ABRIL 2026.xlsx');
let result = {};

workbook.SheetNames.forEach(sheetName => {
  const sheet = workbook.Sheets[sheetName];
  const json = XLSX.utils.sheet_to_json(sheet, { header: 1 }); // read as array of arrays
  // Get first 15 rows to see structure
  result[sheetName] = json.slice(0, 15);
});

fs.writeFileSync('excel_data.json', JSON.stringify(result, null, 2));
